e2e-tests/benchmark.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -Eexuo pipefail
+
+BENCH_OUTPUT=$(cargo bench)
+
+set +e
+REGRESSIONS=$( echo "$BENCH_OUTPUT" |  grep -c "regressed by" )
+set -e
+
+if [[ $REGRESSIONS != 0 ]]; then
+  echo "FAIL! Performance regressions are detected. 
+        Please run $(cargo bench -- --persist) to update $(results.yml)"
```

Comment:
Is wrapping `cargo bench` in a `$(..)` desirable? We don't actually want to run this command as part of emitting the log. ditto for `results.yml`.

Reply:
You are right, I change it.